File: 2418-sort-the-people/2418-sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        name_height = []
        for n, h in zip(names, heights):
            name_height.append((n, h))
        
        return [i for i, j in sorted(name_height, key=lambda x:x[1], reverse=True)]

File: 2418-sort-the-people/2418-sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        vector<bool> visited(names.size());
        vector<string> sorted;
        
        int maxHeight, maxHeightIdx;
        for(int i = 0; i < names.size(); i++) {
            maxHeightIdx = -1;
            maxHeight = 0;
            for(int i = 0; i < names.size(); i++) {
                if(visited[i] == false && heights[i] > maxHeight) {
                    maxHeight = heights[i];
                    maxHeightIdx = i;
                }
            }
            
            if(maxHeightIdx == -1) break;
            visited[maxHeightIdx] = true;
            sorted.push_back(names[maxHeightIdx]);
            
        }
        
        return sorted;
    }
};